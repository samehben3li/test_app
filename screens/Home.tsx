import { View } from "react-native";
import { homeStyles as styles } from "../styles";
import { Header } from "../components";
import { flagIcon, recentIcon } from "../assets";
import Button from "../components/Button";

interface Props {
  navigation: any;
}

export default function HomeScreen({ navigation }: Props) {
  return (
    <View style={styles.container}>
      <Header home={true} />
      <View style={styles.main}>
        <Button
          style={styles.image}
          icon={flagIcon}
          fn={() => navigation.navigate("createFlag")}
          variant="home"
          globalStyles={styles}
        />
        <Button
          style={styles.image2}
          icon={recentIcon}
          fn={() => navigation.navigate("recentFlags")}
          variant="home"
          globalStyles={styles}
        />
      </View>
    </View>
  );
}
