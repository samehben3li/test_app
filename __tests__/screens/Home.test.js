import React from "react";
import renderer from "react-test-renderer";
import Home from "../../screens/Home";
import { MockedProvider } from "@apollo/client/testing";
import AuthProvider from "../../context/authContext";

describe("Home Screen", () => {
  it("renders without crashing", () => {
    const rendered = renderer
      .create(
        <MockedProvider>
          <AuthProvider value={{}}>
            <Home />
          </AuthProvider>
        </MockedProvider>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
