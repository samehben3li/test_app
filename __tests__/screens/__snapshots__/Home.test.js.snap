// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fafafa",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#fff",
          "borderBottomEndRadius": 40,
          "borderBottomStartRadius": 40,
          "elevation": 24,
          "flexDirection": "row",
          "height": 150,
          "justifyContent": "space-between",
          "paddingBottom": 30,
          "paddingHorizontal": 40,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 12,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 16,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 52,
            "padding": 0,
            "width": 120,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 40,
              "resizeMode": "contain",
              "width": "auto",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "medium",
              "fontSize": 11,
              "height": 13,
              "padding": 0,
              "position": "relative",
              "textAlign": "left",
              "width": "100%",
            }
          }
        >
          Tomato Trolley
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 30,
            "marginBottom": 14,
            "width": 40,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": "100%",
              "resizeMode": "contain",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 6,
          "elevation": 15,
          "flexDirection": "row",
          "height": 70,
          "justifyContent": "center",
          "marginVertical": 6,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 9.51,
          "width": 220,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
            "width": 25,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "bold",
            "fontSize": 11,
            "marginLeft": 10,
          }
        }
      >
        NEW FLAG
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 6,
          "elevation": 15,
          "flexDirection": "row",
          "height": 70,
          "justifyContent": "center",
          "marginVertical": 6,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 9.51,
          "width": 220,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 22,
            "resizeMode": "contain",
            "width": 22,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "bold",
            "fontSize": 11,
            "marginLeft": 10,
          }
        }
      >
        NEW FLAG
      </Text>
    </View>
  </View>
</View>
`;
